ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+        case simdjson::dom::element_type::OBJECT :
+            zval obj;
+            object_init(&obj);
+
+            for (simdjson::dom::key_value_pair field : simdjson::dom::object(element)) {
+                zval value = create_object(field.value);
+                add_property_zval_ex(&obj, field.key.data(), strlen(field.key.data()), &value);
+            }
+            v = obj;
+            break;
+        default:
+            break;
+    }
+    return v;
+}
+
+bool cplus_simdjson_is_valid(const char *json, size_t len)
+{
+    simdjson::dom::element doc;
+    auto error = parser.parse(json, len).get(doc);
+    if (error) {
+        return false;
+    }
+    return true;
+}
+
+void cplus_simdjson_parse(const char *json, size_t len, zval *return_value, unsigned char assoc, u_short depth)
+{
+    simdjson::dom::element doc;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    if (assoc) {
+        *return_value = create_array(doc);
+    } else {
+        *return_value = create_object(doc);
+    }
+}
+
+void cplus_simdjson_key_value(
+	const char *json, size_t len, const char *key, zval *return_value, unsigned char assoc, u_short depth
+) {
+    simdjson::dom::element doc;
+    simdjson::dom::element element;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    error = doc.at(key).get(element);
+
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    if (assoc) {
+        *return_value = create_array(element);
+    } else {
+        *return_value = create_object(element);
+    }
+}
+
+u_short cplus_simdjson_key_exists(const char *json, size_t len, const char *key, u_short depth)
+{
+    simdjson::dom::element doc;
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        return SIMDJSON_PARSE_KEY_NOEXISTS;
+    }
+    error = doc.at(key).error();
+    if (error) {
+        return SIMDJSON_PARSE_KEY_NOEXISTS;
+    }
+    return SIMDJSON_PARSE_KEY_EXISTS;
+}
+
+void cplus_simdjson_key_count(const char *json, size_t len, const char *key, zval *return_value, u_short depth)
+{
+    simdjson::dom::element doc;
+    simdjson::dom::element element;
+
+    auto error = build_parsed_json_cust(doc, json, len, true, depth);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    error = doc.at(key).get(element);
+    if (error) {
+        zend_throw_error(php_json_exception_ce, simdjson::error_message(error), 0);
+        RETURN_THROWS();
+    }
+
+    zval v;
+    switch (element.type()) {
+        //ASCII sort
```

Comment:
What does this comment mean? Was it copied from elsewhere?

Reply:
No idea :D This file is 99% the same as the original one, I only changed the thrown exception because it used to be SPL's `RuntimeException`.